### PR TITLE
DAOS-8412 test: Readded single server support for autotest.

### PR DESCRIPTION
**Pull request by chrisphoffman**
Added support for skipping steps on containers
using prop RF1 or RF2 if domain_nr is less than
required replicas.

Signed-off-by: Christopher Hoffman <christopherx.hoffman@intel.com>

### CODE REVIEW

**Review by saurabhtandan — APPROVED** [review]
LGTM.
Please inform the gatekeeper to re-enable autotest and remove ticket DAOS-7946 from the skip list.

File: src/utils/daos_autotest.c [hunk 1]
```diff
@@ -778,7 +802,79 @@ kv_insertaux(void)
...
+	step_fail("failed to open object: %s", d_errdesc(rc));
+	return -1;
+
+fail_insert:
+	step_fail("failed to insert: %s", d_errdesc(put_rc));
+	return -1;
+
+fail_close:
+	step_fail("failed to close object: %s", d_errdesc(rc));
+	return -1;
```
Comment (dmiter) [inline src/utils/daos_autotest.c:822]: Maybe better to keep only one exit label and move error messages near to goto with if condition to easily verify the message is corresponds the error which happens. I think it's difficult to maintain many labels and always go down to see the error message which prints.

File: src/utils/daos_autotest.c [hunk 2]
```diff
@@ -778,7 +802,79 @@ kv_insertaux(void)
...
+	step_fail("failed to open object: %s", d_errdesc(rc));
+	return -1;
+
+fail_read:
+	step_fail("failed to read: %s", d_errdesc(get_rc));
+	return -1;
+
+fail_close:
+	step_fail("failed to close object: %s", d_errdesc(rc));
+	return -1;
```
Comment (dmiter) [inline src/utils/daos_autotest.c:865]: the same here.

**Review by dmiter — COMMENTED** [review]
I'm fine in general but I'd like to avoid many exit labels and move error messages near to goto with if condition.

**Review by dmiter — APPROVED** [review]
This looks much better to me. :-) Thanks!

File: src/utils/daos_autotest.c (hunk 1 above)
Comment (chrisphoffman) [reply src/utils/daos_autotest.c]: Great comments and feedback. I'm always interested in increasing maintainability and readability in code. This should be all set

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by jolivier23 — APPROVED** [review]

**Comment by jolivier23** [comment]
Please add daos-gatekeeper again when it is finished

**Pull request merged**